### ISSUE TITLE
Wrong stake amount after partial sell if canceled on exchange while only partly-filled

### ISSUE THREAD

**Issue by VanessaE** [freqtrade/freqtrade#7444]
## Describe your environment

  * Operating system: Debian "buster" 11.5
  * Python Version: 3.9.2
  * CCXT version: 1.93.66 (may not be correct!  I updated freqtrade before filing this issue)
  * Freqtrade Version: `develop` branch, git commit 93cbfc5f294afc4e89e73d0161d024bf07ad6cf3 (this bit IS correct)
  
## Describe the problem:

After a series of actions that seemed normal to me at first, I ended up with a trade that freqtrade could not sell, because it thought it was supposed to sell a larger stake than it actually had.  In the name of privacy, the name and quantities of the coin in question have been obfuscated, this being a public venue and all.  That said, the stake amounts involved were well in excess of the exchange's minimum limits, so that isn't the cause.

To start off, the strategy bought 200 COIN through its normal process, i.e. just going about its usual business of buying and selling.  It held onto this trade for about a day before I stuck my nose in. :stuck_out_tongue:  

I decided I needed some money for something else, so I did a partial sell of 50 COIN.  The partial sell filled normally, and generated "exiting" and "exited" notifications in Telegram, as usual.  This partial sell left 150 COIN, exactly as expected.

Immediately after, I realized I needed a bit more money for that other thing, so I did another partial sell of 50 COIN.  The bot generated an "exiting" notification for this action, as well.

That order was taking forever to fill, but it did get far enough along to meet my immediate need, so I canceled the order via the exchange with 10 COIN left to go, leaving the trade with a final stake of 110 COIN.

Canceling the order there caused freqtrade to issue a "canceled" notification, just expected, but the partial sell action itself \*did not\* generate the "exited" notification like with the first partial sell.  I didn't initially notice this, though (not until later, when I skimmed through Telegram in pursuit of this issue).

Later, when I saw that COIN was just coming to the end of a nice pump, I knew the strategy would probably miss the optimal exit, so I took profit and tried to close-out that trade entirely.  That was with a routine force exit, not a partial sell, same as any other day.

That should have issued a sell order for 110 COIN, but the bot did not generate an "exiting" notification, which prompted me to check the log.  Therein, I saw that the bot was trying to sell *150* COIN, i.e. the amount that remained after the first partial sell filled, thus it was unable to close the trade:

```
2022-09-19 00:24:09,298 - freqtrade.freqtradebot - INFO - Exit for COIN/XYZ detected. Reason: custom_exit
2022-09-19 00:24:09,519 - freqtrade.wallets - INFO - Wallets synced.
2022-09-19 00:24:09,519 - freqtrade.freqtradebot - WARNING - Unable to exit trade COIN/XYZ: Not enough amount to exit trade. Trade-amount: 150, Wallet: 110
2022-09-19 00:24:18,822 - freqtrade.worker - INFO - Bot heartbeat. PID=305215, version='2022.9.dev-93cbfc5f2, strategy_version: v11.2.140', state='RUNNING'
```

I did a `reload_config` to try to prod the bot to correct the stake amount and close the trade, but that didn't help.  I had to delete the trade entirely and sell it via the exchange.

There were no rebuys on this particular trade.  Just the initial buy and the above-described sells.

### Steps to reproduce:

I am not sure if this is repeatable....

1. Buy something, preferably some low-volume coin that takes a long time to fill.
2. Hold it for a while, maybe even just 5 minutes.
3. Do a partial sell via frequi.
4. Do another partial sell via frequi.
5. Go on the exchange and cancel that order *before* it can completely fill.
6. Hold the remainder for a while, say another 5 minutes.
7. Do a final force-exit via frequi.

### Observed Results:

  * What happened?
  * What did you expect to happen?

The bot lost track of how much it could actually sell, and thus could not close-out the trade.  Obviously, it should have tracked the amount properly at all times :smiley:, but perhaps as a backup, it should automatically use the amount the exchange wallet actually has in it, if that's less than the assumed stake amount.

### Relevant code exceptions or logs

Just before the bot went down for the aforementioned restart, it threw this python error:

<details><summary><b>First python error...</b></summary>

```
2022-09-18 23:29:05,628 - uvicorn.error - ERROR - Exception in ASGI application
Traceback (most recent call last):
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 404, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/uvicorn/middleware/message_logger.py", line 86, in __call__
    raise exc from None
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/uvicorn/middleware/message_logger.py", line 82, in __call__
    await self.app(scope, inner_receive, inner_send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/fastapi/applications.py", line 269, in __call__
    await super().__call__(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/middleware/cors.py", line 92, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/middleware/cors.py", line 147, in simple_response
    await self.app(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/exceptions.py", line 93, in __call__
    raise exc
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
    await self.app(scope, receive, sender)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/routing.py", line 670, in __call__
    await route.handle(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/routing.py", line 266, in handle
    await self.app(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/routing.py", line 65, in app
    response = await func(request)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/fastapi/routing.py", line 231, in app
    raw_response = await run_endpoint_function(
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/fastapi/routing.py", line 162, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/rpc/api_server/api_v1.py", line 164, in forceexit
    return rpc._rpc_force_exit(payload.tradeid, ordertype, amount=payload.amount)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/rpc/rpc.py", line 733, in _rpc_force_exit
    self.__exec_force_exit(trade, ordertype, amount)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/rpc/rpc.py", line 702, in __exec_force_exit
    self._freqtrade.execute_trade_exit(
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/freqtradebot.py", line 1560, in execute_trade_exit
    self._notify_exit(trade, order_type, sub_trade=bool(sub_trade_amt), order=order_obj)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/freqtradebot.py", line 1622, in _notify_exit
    self.rpc.send_msg(msg)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/rpc/rpc_manager.py", line 77, in send_msg
    mod.send_msg(msg)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/rpc/telegram.py", line 441, in send_msg
    message = self.compose_message(msg, msg_type)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/rpc/telegram.py", line 382, in compose_message
    message = self._format_exit_msg(msg)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/rpc/telegram.py", line 355, in _format_exit_msg
    message += (f"*Current Rate:* `{msg['current_rate']:.8f}`\n"
TypeError: unsupported format string passed to NoneType.__format__
```
</details>

Digging through the logs as requested, I found nothing that suggested a fault with opening either of the sell orders, but the bot did keep repeating the aforementioned stake amount vs. wallet warning every so often.

In the midst of those repeated warnings, I found this python error:

<details><summary><b>Second python error...</b></summary>

```
2022-09-19 14:01:51,823 - freqtrade.freqtradebot - WARNING - Unable to exit trade COIN/XYZ: Not enough amount to exit trade. Trade-amount: 150, Wallet: 110
2022-09-19 14:01:53,524 - freqtrade.wallets - INFO - Wallets synced.
2022-09-19 14:01:53,525 - uvicorn.error - ERROR - Exception in ASGI application
Traceback (most recent call last):
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 404, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/uvicorn/middleware/message_logger.py", line 86, in __call__
    raise exc from None
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/uvicorn/middleware/message_logger.py", line 82, in __call__
    await self.app(scope, inner_receive, inner_send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/fastapi/applications.py", line 269, in __call__
    await super().__call__(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/middleware/cors.py", line 92, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/middleware/cors.py", line 147, in simple_response
    await self.app(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/exceptions.py", line 93, in __call__
    raise exc
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
    await self.app(scope, receive, sender)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/routing.py", line 670, in __call__
    await route.handle(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/routing.py", line 266, in handle
    await self.app(scope, receive, send)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/routing.py", line 65, in app
    response = await func(request)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/fastapi/routing.py", line 231, in app
    raw_response = await run_endpoint_function(
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/fastapi/routing.py", line 162, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/starlette/concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/home/vanessa/freqtrade-related/freqtrade-spot/.env/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/rpc/api_server/api_v1.py", line 164, in forceexit
    return rpc._rpc_force_exit(payload.tradeid, ordertype, amount=payload.amount)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/rpc/rpc.py", line 733, in _rpc_force_exit
    self.__exec_force_exit(trade, ordertype, amount)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/rpc/rpc.py", line 702, in __exec_force_exit
    self._freqtrade.execute_trade_exit(
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/freqtradebot.py", line 1515, in execute_trade_exit
    amount = self._safe_exit_amount(trade.pair, sub_trade_amt or trade.amount)
  File "/home/vanessa/freqtrade-related/freqtrade-spot/freqtrade/freqtradebot.py", line 1463, in _safe_exit_amount
    raise DependencyException(
freqtrade.exceptions.DependencyException: Not enough amount to exit trade. Trade-amount: 150, Wallet: 110

```
</details>

I found nothing else in the log that suggested there was a problem.

**Comment by moinonin** [comment]
Just close this trade manually by accessing the database and type:

```
update trades set is_open=0 where id={get trade id from telegram}
```
You may also want to remove from orders:
```
delete from orders where order_id={get order_id from logs}
```

**Comment by stash86** [comment]
or they can just delete the trade from telegram easily, but that's not the point here

**Comment by xmatthias** [comment]
partially filled exit orders are currently not supported.

Freqtrade would never cancel this (there's currently no logic to handle partially filled exits).
Now obviously we can't prevent users from canceling such an order on exchange - but that still doesn't change the fact that there's no logic to handle this case.

To some degree, this makes this a duplicate of #3009. 
While that issue is around "freqtrade won't cancel my exit" - the point is really the same - we currently don't handle partially filled exits.

**Comment by xmatthias** [comment]
The "real" interesting thing would be the log messages from around the time the partial (and partially filled) sell order was canceled.

Feel free to either share here or via DM.

**Comment by VanessaE** [comment]
(detailed logs relayed via DM)

**Comment by VanessaE** [comment]
It seems that any partial sell order has the potential to "succeed silently"...

I had it happen with another trade after filing this issue, and THAT one was completely routine -- no cancellations, no slow fills, no funny business.  Just the usual entry, a few "rebuys", and a couple of partial sell attempts.

No database entries were created for the sell orders, and no Telegram notifications ware generated, yet the orders fully filled on the exchange, got recorded into the ledger there, and the open trade was left showing a larger stake amount than it should have, exactly like my initial issue.

**Comment by xmatthias** [comment]
> No database entries were created for the sell orders, and no Telegram notifications ware generated, yet the orders fully filled on the exchange, got recorded into the ledger there, and the open trade was left showing a larger stake amount than it should have, exactly like my initial issue.

without logs - this will be a tricky one to even attempt to debug.

**edit** i now understand that you think it's the same issue ...
